airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigPersistence.java

```diff
@@ -37,4 +37,6 @@
 
   <T> void writeConfig(ConfigSchema configType, String configId, T config) throws JsonValidationException, IOException;
 
+  void deleteOrphanDirectories() throws IOException;
```

Comment:
`ConfigPersistence` keep in mind we want to create a `PostgresConfigPersistence`. surfacing things about directories in this interface ties the interface specifically to a single implementation (the file system one). I think you can either have a generic `cleanUpOldTables` function or something, but it feels kinda awkward to me. That's why I was suggesting the atomic replace here because it solves a transaction problem that already exists with the interface and gets rid of this weirdness. I think it's fine to replace part of the ImportArchive implementation with usage of atomic replace. the super verbose thing we have right now is pretty error prone anyway.